python/mxnet/test_utils.py

```diff
@@ -116,7 +116,22 @@ def _validate_csr_generation_inputs(num_rows, num_cols, density,
                              % (density, num_rows, num_cols))
 
 
-def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None):
+def shuffle_csr_column_indices(csr):
+    """Shuffle CSR column indices per row
+    This allows validation of unordered column indices, which is not a requirement
+    for a valid CSR matrix
+    """
+    row_count = len(csr.indptr) - 1
+    for i in range(row_count):
+        start_index = csr.indptr[i]
+        end_index = csr.indptr[i + 1]
+        sublist = np.array(csr.indices[start_index : end_index])
+        np.random.shuffle(sublist)
+        csr.indices[start_index : end_index] = sublist
+
+
+def _get_uniform_dataset_csr(num_rows, num_cols, density=0.1, dtype=None,
```

Comment:
If it's not supported, I don't feel it's a good idea to keep this function disclosed to users in the release. It may convey misleading messages that we support unordered column indices in the release. On the other hand, csr with unordered column indices would greatly affect the performance of not only dot operator we have implemented, but also element-wise operators such as add, subtract, multiply, etc we are going to implement. Before we figure out the necessity of supporting that, it's better to keep the release scope as small as possible.